### DISCUSSION TITLE
How to use SlackClientHttpConnector for SlackClient?

### DISCUSSION THREAD

**Discussion by akash-osec** [abdolence/slack-morphism-rust#166] [Q&A] [answered]
let client = SlackClient::new(<SlackClientHttpConnector>);
    let token_value: SlackApiTokenValue = "xoxb---".into();
    let token: SlackApiToken = SlackApiToken::new(token_value);
    let session = client.open_session(&token);

    let post_chat_req =
     SlackApiChatPostMessageRequest::new("#reports".into(),
            SlackMessageContent::new().with_text("Hey there! pinged from rust".into())
    );

    let post_chat_resp = session.chat_post_message(&post_chat_req).await?;

**Comment by abdolence** [comment, marked as answer]
Hey, 

`SlackClientHttpConnector` is designed to be able to implement different HTTP connectors and use different async runtimes. I know people were interested to have/implement `async-std` themselves for example.

Saying that this crate provides only default implementation for Hyper/Tokio (so I don't support two runtimes myself in the crate itself), so if you're not going to implement HTTP backend/provider for your specific needs you just need to enable the one that crate provides just enabling the feature in your deps:
```
slack-morphism = { version = "1.4", features = ["hyper"] }
```
and using it like this:

```rust
let client = SlackClient::new( SlackClientHyperConnector::new() );
```

As described in the docs: 
https://slack-rust.abdolence.dev/web-api.html
